util/devel/mason-completion.bash

```diff
@@ -0,0 +1,275 @@
+_complete_mason_doc(){
+  COMPREPLY=($(compgen -W "--help -h" -- "$1" ))
+}
+
+_complete_mason_clean(){
+  COMPREPLY=($(compgen -W "--help -h" -- "$1" ))
+}
+
+_complete_mason_env(){
+  COMPREPLY=($(compgen -W "--help -h" -- "$1" ))
+}
+
+_complete_mason_search(){
+  _options search
+}
+
+_complete_mason_update(){
+  local cur
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in
+    -*) 
+      _options update
+      ;;
+     *)
+      ;;
+  esac
+}
+
+_complete_mason_run(){
+  local optional
+  local optional_abbr
+  optional=$(mason run --help | grep -v '^[A-Za-z]' | grep '^.*--' |      \
+            sed -e 's;^\(.*--[^ ]*\).*$;\1;' | sed -e 's;-[A-Za-z],;   ;' | \
+             sed -e 's/.*=//g' |\
+             xargs echo)
+  optional_abbr=$(mason run --help | grep -v '^[A-Za-z]' | grep '^.*--' |        \
+          sed -e 's;^\(.*--[^ ]*\).*$;\1;' | grep ',' | sed -e 's;,.*$;;' | \
+          xargs echo)
+  local cur
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in
+    -*)
+      COMPREPLY=($(compgen -W "$optional $optional_abbr" -- "${COMP_WORDS[COMP_CWORD]}" ))
+      ;;
+    *)
+      COMPREPLY=($(compgen -d -S / -- "$cur"))
+      ;;
+  esac
+}
+
+_complete_mason_build(){
+  local optional
+  local optional_abbr
+  optional=$(mason build --help | grep -v '^[A-Za-z]' | grep '^.*--' |      \
+            sed -e 's;^\(.*--[^ ]*\).*$;\1;' | sed -e 's;-[A-Za-z],;   ;' | \
+             sed -e 's/.*=//g' | sed -e 's/[][]//g'|  \
+             xargs echo)
+  optional_abbr=$(mason build --help | grep -v '^[A-Za-z]' | grep '^.*--' |        \
+          sed -e 's;^\(.*--[^ ]*\).*$;\1;' | grep ',' | sed -e 's;,.*$;;' | \
+          xargs echo)
+  local cur
+  local update_tag="--update"
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in 
+    -*) 
+      COMPREPLY=($(compgen -W "$optional $optional_abbr $update_tag" -- "${COMP_WORDS[COMP_CWORD]}" ))
+      ;;
+     *)
+      ;;
+  esac
+}
+
+_complete_mason_add(){
+  _options add
+}
+
+_complete_mason_init(){
+  local cur
+  cur=${COMP_WORDS[COMP_CWORD]}
+  case "$cur" in
+    -*)
+      _options init 
+      ;;
+    *)
+      COMPREPLY=($(compgen -d -S / -- "$cur"))
+      ;;
+  esac
+}
+
+_complete_mason_new(){
+  _options new
+}
+_complete_mason_option(){
+  local optional
+  local optional_abbr
+  optional=$(mason --help | grep -v '^[A-Za-z]' | grep '^.*--' |      \
+            sed -e 's;^\(.*--[^ ]*\).*$;\1;' | sed -e 's;-[A-Za-z],;   ;' | \
+            xargs echo)
+  optional_abbr=$(mason --help | grep -v '^[A-Za-z]' | grep '^.*--' |        \
+          sed -e 's;^\(.*--[^ ]*\).*$;\1;' | grep ',' | sed -e 's;,.*$;;' | \
+          xargs echo)
+  COMPREPLY=($(compgen -W "$optional $optional_abbr" -- "$1" ))
+}
+
+_complete_mason_command(){
+  local cur
+  local prev
+  local arg
+  cur=${COMP_WORDS[COMP_CWORD]}
+  prev="${COMP_WORDS[COMP_CWORD-1]}"
+  arg=$(mason --help | grep -v '^[A-Za-z]' | grep -v '^.*--'  | \
+     grep -v '<command>' | grep -v '\[options\]'  | sed 's/^[ \t]*//' | \
```

Comment:
The `\t` in `sed 's/^[ \t]*//' ` is removing `t` character in the MacOS (BSD) version of sed.

Reply:
Based on [this SO post](https://stackoverflow.com/a/38023442/6771423) I believe we can fix this by using ANSI-C quote style:

```suggestion
     grep -v '<command>' | grep -v '\[options\]'  | sed $'s/^[ \t]*//' | \
```

If this works on your machine, do you want to switch all the sed commands to this mode?